crates/common_utils/src/new_type.rs

```diff
@@ -1,10 +1,203 @@
 //! Contains new types with restrictions
-use crate::consts::MAX_ALLOWED_MERCHANT_NAME_LENGTH;
+use crate::{consts::MAX_ALLOWED_MERCHANT_NAME_LENGTH, pii::UpiVpaMaskingStrategy};
+use masking::{PeekInterface, Secret, SerializableSecret};
 
 #[nutype::nutype(
     derive(Clone, Serialize, Deserialize, Debug),
     validate(len_char_min = 1, len_char_max = MAX_ALLOWED_MERCHANT_NAME_LENGTH)
 )]
 pub struct MerchantName(String);
 
-impl masking::SerializableSecret for MerchantName {}
+impl SerializableSecret for MerchantName {}
+
+/// Function for masking alphanumeric characters in a string.
+///
+/// # Arguments
+///     `val`
+///         - holds reference to the string to be masked.
+///     `unmasked_char_count`
+///         - minimum character count to remain unmasked for identification
+///         - this number is for keeping the characters unmasked from
+///             both beginning (if feasible) and ending of the string.
+///     `min_masked_char_count`
+///         - this ensures the minimum number of characters to be masked
+///
+/// # Info
+///     In case the sum of `unmasked_char_count` and `min_masked_char_count` exceeds
+///     the total length of the string, `unmasked_char_count` is considered over `min_masked_char_count`.
```

Comment:
right, but the doc comment is contradicting this condition isn't it? in case the sum exceeds total length, algo considers 0 as start index instead of `unmasked_char_count`. Or am I missing something here?

Reply:
Updated the doc comment! It should be clear now.

I was trying to explain that `unmasked_char_count` is always considered over `min_masked_char_count` for cases where the sum of this exceeds the string length.

For instance, I can have a string `0ABFE7198FD` and I can say

`apply_mask(val, 4, 100)` - meaning I want to keep 4 characters unmasked but minimum masked characters should be 100 - which is not possible for this value, so the modified string will have only the last 4 characters unmasked, and rest of them will be masked